### PR TITLE
refactor: creating instance w/o unnecessary configs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

creating instance w/o unnecessary configs

## Testing

Tested locally.

### CODE REVIEW

File: packages/connector-saml/src/utils.ts [hunk 1]
```diff
@@ -61,17 +62,17 @@ export const samlAssertionHandler = async (
   // eslint-disable-next-line new-cap
   const identityProvider = saml.IdentityProvider({
     metadata: idpMetadataXml,
-    messageSigningOrder,
     isAssertionEncrypted: encryptAssertion,
+    messageSigningOrder,
```
Comment (gao-sun) [inline packages/connector-saml/src/utils.ts:66]: this is not chore. Like a refactor

File: packages/connector-saml/src/index.ts [hunk 2]
```diff
@@ -75,15 +75,15 @@ const getAuthorizationUri =
...
-        privateKeyPass,
         assertionConsumerService: [
           {
             Location: assertionConsumerServiceUrl,
             Binding: saml.Constants.BindingNamespace.Post,
           },
         ],
+        requestSignatureAlgorithm,
+        privateKey: conditional(signAuthnRequest && privateKey),
+        privateKeyPass: conditional(signAuthnRequest && privateKeyPass),
```
Comment (charIeszhao) [inline packages/connector-saml/src/index.ts:86]:
```ts
...conditional(signAuthnRequest && {
  privateKey,
  privatekeyPass,
})
```

**Review by charIeszhao — APPROVED** [review]
OLGTM 😎

File: packages/connector-saml/src/index.ts (hunk 2 above)
Comment (darcyYe) [reply packages/connector-saml/src/index.ts]: updated

File: packages/connector-saml/src/utils.ts (hunk 1 above)
Comment (gao-sun) [reply packages/connector-saml/src/utils.ts:66]: this is actually a bug fix?

**Pull request closed without merging**